fpga_interchange/arch.h

```diff
@@ -696,12 +752,14 @@ struct Arch : ArchAPI<ArchRanges>
     std::unordered_map<WireId, std::pair<int, int>> driving_pip_loc;
     std::unordered_map<WireId, NetInfo *> reserved_wires;
 
+    static constexpr size_t kMaxState = 8;
     struct TileStatus
     {
+        std::vector<ExclusiveStateGroup<kMaxState>> tags;
         std::vector<CellInfo *> boundcells;
     };
 
-    std::vector<TileStatus> tileStatus;
+    std::unordered_map<int32_t, TileStatus> tileStatus;
```

Comment:
I'm kind of curious of the performance tradeoff here

Reply:
I believe for smaller designs, this will be good. I think for large fabrics, this will need to be revisited.  Because of how I moved access to the `tileStatus` behind `get_tile_status` for all of the mutable cases, I believe this will be easy to change as needed once we have more test data.